viper/parser/stmt.py

```diff
@@ -204,9 +184,11 @@ def parse_for(self):
             start = Expr.parse_value_expr(self.stmt.iter.args[0], self.context)
             rounds = self.stmt.iter.args[1].right.n
         varname = self.stmt.target.id
-        pos = self.context.vars[varname].pos if varname in self.context.forvars else self.context.new_variable(varname, BaseType('num'))
-        o = LLLnode.from_list(['repeat', pos, start, rounds, parse_body(self.stmt.body, self.context)], typ=None, pos=getpos(self.stmt))
+        pos = self.context.new_variable(varname, BaseType('num'))
```

Comment:
if you create new variable for each index variable, it will reject program like:
```
def foo():
    for i in range(10):
        pass
    for i in range(20):
        pass
```
Is it intended?

Reply:
It doesn't reject it because although I'm creating a new index variable every time I'm also deleting it at the end of the for loop, see [here](https://github.com/DavidKnott/viper/blob/b163146e5171b181b716a7681f903cf6562d014e/viper/parser/stmt.py#L271).

Reply:
i see. I didn't read the code carefully. Now, the scope of the index variable is the loop block instead of the whole function.

Reply:
wait. You may also want to remove the index variable from self.context.forvars. Otherwise, the following program will report weird error:
```
@public
def foo():
    for i in range(10):
        pass
    i = 100  // create new variable i
    i = 200  // look up the variable i and check whether it is in forvars
```